pkg/apis/recognizer/devfile_recognizer.go

```diff
@@ -45,11 +45,29 @@ func SelectDevFilesFromTypes(path string, devfileTypes []model.DevfileType) ([]i
 	if err != nil {
 		return []int{}, err
 	}
-	devfile, err := SelectDevfileUsingLanguagesFromTypes(languages, devfileTypes)
+	mainLanguage, err := getMainLanguage(languages)
+	if err != nil {
+		return []int{}, err
+	}
+	devfiles, err := selectDevfilesByLanguage(mainLanguage, devfileTypes)
 	if err != nil {
 		return []int{}, errors.New("No valid devfile found for project in " + path)
 	}
-	return []int{devfile}, nil
+	return devfiles, nil
```

Comment:
Do we need to check for the length of the devfile index first since that's what `selectDevfilesByLanguage()` returns?

Reply:
IMO I think is not required. If I understand correctly each time we have an error we already return an empty list, so for empty int list (no devfile index matched) is ok to return it.